src/gui_common/TweakedColourPicker.cs

```diff
@@ -0,0 +1,357 @@
+        }
+    }
+
+    [Export]
+    public new bool HsvMode
+    {
+        get => base.HsvMode;
+        set
+        {
+            if (value && RawMode)
+                return;
+
+            base.HsvMode = value;
+            UpdateButtonsState();
+
+            // Update HSV sliders' tooltips
+            UpdateTooltips();
+        }
+    }
+
+    [Export]
+    public new bool RawMode
+    {
+        get => base.RawMode;
+        set
+        {
+            if (value && HsvMode)
+                return;
+
+            base.RawMode = value;
+            UpdateButtonsState();
+        }
+    }
+
+    [Export]
+    public new bool PresetsEnabled
+    {
+        get => presetsEnabled;
+        set
+        {
+            presetsEnabled = value;
+
+            if (addPresetButton == null)
+                return;
+
+            addPresetButton.Disabled = !value;
+        }
+    }
+
+    [Export]
+    public new bool PresetsVisible
+    {
+        get => presetsVisible;
+        set
+        {
+            presetsVisible = value;
+
+            if (presetsContainer == null)
+                return;
+
+            separator.Visible = value;
+            presetsContainer.Visible = value;
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        // Hide replaced native controls. Can't delete them because it will crash Godot.
+        GetChild(4).GetChild<Control>(4).Hide();
+        GetChild<Control>(5).Hide();
+        GetChild<Control>(6).Hide();
+        GetChild<Control>(7).Hide();
+
+        // Get controls
+        var baseControl = GetChild(4);
+        sliderROrH = baseControl.GetChild(0).GetChild<HSlider>(1);
+        sliderGOrS = baseControl.GetChild(1).GetChild<HSlider>(1);
+        sliderBOrV = baseControl.GetChild(2).GetChild<HSlider>(1);
+        sliderA = baseControl.GetChild(3).GetChild<HSlider>(1);
+        pickerButton = GetChild(1).GetChild<ToolButton>(1);
+        hsvCheckButton = GetNode<CheckButton>("MarginButtonsContainer/ButtonsContainer/HSVCheckButton");
+        rawCheckButton = GetNode<CheckButton>("MarginButtonsContainer/ButtonsContainer/RawCheckButton");
+        htmlColourEdit = GetNode<LineEdit>("MarginButtonsContainer/ButtonsContainer/HtmlColourEdit");
+        separator = GetNode<HSeparator>("Separator");
+        presetsContainer = GetNode<GridContainer>("PresetContainer");
+        addPresetButton = GetNode<TextureButton>("PresetButtonContainer/AddPresetButton");
+
+        // Update control state.
+        UpdateButtonsState();
+        PresetsEnabled = presetsEnabled;
+        PresetsVisible = presetsVisible;
+        OnColourChanged(Color);
+
+        // Load presets.
+        if (PresetsStorage.Exists(p => p.Key == GetPath()))
+        {
+            var presetsStored = PresetsStorage.First(p => p.Key == GetPath());
+            foreach (var colour in presetsStored.Value)
+                AddPreset(colour);
+            PresetsStorage.Remove(presetsStored);
```

Comment:
I think the presets shouldn't be cleared. Only overwritten when this exits properly. This would guard against exiting in a weird way that loses the presets.